Modelica/Media/R134a.mo

```diff
@@ -326,19 +326,27 @@ rho = Medium.density(setState_phX(p, h, fill(0, Medium.nX)));
 
     algorithm
       R_s := R134aData.data.R_s;
-      f := f_R134a(d=d, T=T);
-      state.p := d*R_s*T*f.delta*f.fdelta;
-      state.h := R_s*T*(f.tau*f.ftau + f.delta*f.fdelta);
-      sat.psat := state.p;
-      state.T := T;
-      state.d := d;
-      sat.Tsat := state.T;
+      sat := setSat_T(T);
       dl := bubbleDensity(sat);
       dv := dewDensity(sat);
-      state.phase := if ((state.d < dl) or (state.d > dv) or (state.p >
-        R134aData.data.FPCRIT)) then 1 else 2;
-
+      if d < dl and d > dv and T < R134aData.data.FTCRIT then
+        f := Modelica.Media.Common.HelmholtzDerivs(
+          d=d, T=T, R_s=R_s, delta=0, tau=0, f=0, fdelta=0,
+          fdeltadelta=0, ftau=0, ftautau=0, fdeltatau=0);
+        state.p := saturationPressure(T);
+        state.h := (dl*dv/d*(dewEnthalpy(sat) - bubbleEnthalpy(sat))
+                  - dv*dewEnthalpy(sat) + dl*bubbleEnthalpy(sat))/(dl - dv);
```

Comment:
I did not check this expression, but it would be more readable using an intermediate variable for the vapour quality.

Reply:
This would not be of help here, since we @casella resolved the vapor quality for h.